sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -543,6 +543,12 @@ object ScalaReflection extends ScalaReflection {
             throw new UnsupportedOperationException(s"`$fieldName` is a reserved keyword and " +
               "cannot be used as field name\n" + walkedTypePath)
           }
+          // It is ok to use head here because a variable name needs to be non empty
+          if (fieldName.head.isDigit) {
+            throw new UnsupportedOperationException(s"`$fieldName` is a word that starts with a " +
+              "number and cannot be used as field name" + "A work around to this problem is to " +
```

Comment:
- `name"` -> `name."`
- `work around` -> `workaround`.

And, I guess we don't need to split there, `field name. A workaround is to convert " +`.

Reply:
Done